### PR TITLE
bpo-33602: Remove set and queue references from Data Types

### PR DESCRIPTION
**Pull request by andresdelfino**
https://bugs.python.org/issue33602

### CODE REVIEW

**Review by serhiy-storchaka — APPROVED** [review]

**Review by rhettinger — DISMISSED** [review]
Please leave "heap queues" and "synchronized queues" as-is.  They lose some of their meaning if you factor out the word "queue" which means something slightly different in each of the three cases.

Go ahead with removing "set" and adding "enumerations"

**Comment by andresdelfino** [comment]
@rhettinger I removed "synchronized queues" because they are now described in Concurrent Execution, and it seemed confusing to mention them if they are not listed below.

I see your point about "heap" without "queue" attached to it. I could omit the mention to double-ended queues, so it reads "heap queues" again.

Would that be ok for you?

**Comment by andresdelfino** [comment]
While I didn't literally made the requested changes, I'm pretty sure I addressed the issue raised and I think there was a confusion between "synchronized queues" and "double-ended queues", so:

I have made the requested changes; please review again

If I was wrong, I'll be glad to amend any error.

File: Doc/library/datatypes.rst [hunk 1]
```diff
@@ -5,13 +5,14 @@ Data Types
 **********
 
 The modules described in this chapter provide a variety of specialized data
-types such as dates and times, fixed-type arrays, heap queues, synchronized
-queues, and sets.
+types such as dates and times, fixed-type arrays, heap queues, double-ended
```
Comment (andresdelfino) [inline Doc/library/datatypes.rst:8]: Description of synchronized queues (the queue module) was moved to the Concurrent Execution section of the Library Reference. Mention double-ended queues instead, which are actually described in this section.

**Comment by andresdelfino** [comment]
@rhettinger any chance you could give a look at the changes made and my comments?

**Comment by andresdelfino** [comment]
@rhettinger could you take a look at the changes made after your comments? Please note the comment I left in https://github.com/python/cpython/pull/7055#pullrequestreview-124235222

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no longer accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

**Comment by andresdelfino** [comment]
@rhettinger could you take a look at this?

**Comment by andresdelfino** [comment]
@serhiy-storchaka could you -as the other reviewer of this PR- take a look on the changes made and see if they address the concerns Raymond shared?

**Comment by JulienPalard** [comment]
(closing and re-opening to restart CI, Azure was stuck)

**Comment by JulienPalard** [comment]
@andresdelfino thanks!

**Pull request merged**